### DISCUSSION TITLE
elliptical inclination control from Zwift

### DISCUSSION THREAD

**Discussion by realericmc** [cagnulein/qdomyos-zwift#2280] [Q&A] [answered]
So qdomyos can control the inclination on my Sole E35 via the buttons on the tile, I have put in credentials for zwift and the zwift treadmill auto inclination setting is on (the first one for android and iOS since I am running it on an iPhone).  It's not working yet though, so what else do I need to do for zwift to control the inclination?  Appreciate any pointers to what I am missing.  thanks!

**Comment by cagnulein** [comment]
Hi share me a debug log showing the issue and I will check it ASAP
https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 7 apr 2024 alle 01:19 Eric McMurry ***@***.***>
ha scritto:

**Reply by realericmc** [reply]
Hi Roberto,  Thanks for taking a look.  Here's a log from earlier today with a zwift session on a route that had elevation changes.  I'm hoping I have just missed a step somewhere.

[debug-Sat_Apr_6_16_24_39_2024.log.zip](https://github.com/cagnulein/qdomyos-zwift/files/14895459/debug-Sat_Apr_6_16_24_39_2024.log.zip)

**Comment by cagnulein** [comment]
Apr 6 16:24:40 2024 1712438680381 Debug:
./../src/zwift-api/zwift_client_auth.h void
AuthToken::handleTokenResponse(QNetworkReply *) Error fetching token: "Host
requires authentication"

Seems a wrong password or username or both

You have to see a green check Mark right to the zwift credentials in qz
after a restart

Let me know

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 7 apr 2024 alle 03:46 Eric McMurry ***@***.***>
ha scritto:

**Reply by realericmc** [reply]
Thanks for checking that.  I reentered the credentials again with the same result and I see the same thing in the log.  There is a '+' in the username.  Could that be causing an issue with what is being passed to zwift?

**Comment by cagnulein** [comment]
Yes could be, try to add a \ before the +

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 7 apr 2024 alle 04:14 Eric McMurry ***@***.***>
ha scritto:

**Reply by realericmc** [reply]
good idea, but still no joy.  Same error.

**Comment by cagnulein** [comment]
I guess I Have to escape the character from the code directly. I will do it
next week 4 AM here in Italy and I'm running.

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 7 apr 2024 alle 04:21 Eric McMurry ***@***.***>
ha scritto:

**Reply by realericmc** [reply]
no rush.  I was surprised you responded at all :-)

Thanks!

**Reply by cagnulein** [reply]
ok @realericmc I tried to do a modification here https://github.com/cagnulein/qdomyos-zwift/actions/runs/8595846020 for you.
Do you have a spare android device to test it?

**Comment by realericmc** [comment]
I believe the elliptical is running android, but I don’t know that you can load apps on it.  Other than that, I do not have any android devices.

Thanks!

**Reply by cagnulein** [reply]
argh okay, so i will send you an ios build later this week

**Reply by realericmc** [reply]
Thank @cagnulein .  I  tried to give it a go, but it's been a bit since I did an iOS build and fixing the cert stuff took more time than I had.

**Reply by cagnulein** [reply]
@realericmc no i will send you it, you don't have to do anything. i will have a free slot this thursday

**Reply by cagnulein** [reply, marked as answer]
@realericmc done, i don't know if it would work but let's give it a try. 
send me an email to roberto.viola83 at gmail.com indicating this ticket and if you are on android or iOS. Thanks

**Reply by realericmc** [reply]
works!  green checkmark and the elevation changes in zwift

Thanks @cagnulein!

**Comment by cagnulein** [comment]
Amazing!
remember to rate the app on the store if you didn't already ! It means a
lot to me!
Thanks!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 18 apr 2024 alle 18:15 Eric McMurry ***@***.***>
ha scritto:

**Reply by realericmc** [reply]
done. thanks again for all the changes to get this going!
